src/components/camera.js

```diff
@@ -23,22 +23,25 @@ module.exports.Component = registerComponent('camera', {
    * Add reference from scene to this entity as the camera.
    */
   init: function () {
-    var camera = this.camera = new THREE.PerspectiveCamera();
-    var sceneEl = this.el.sceneEl;
-    this.el.setObject3D('camera', camera);
-    this.bindMethods();
-    sceneEl.addEventListener('enter-vr', this.removeHeightOffset);
-    sceneEl.addEventListener('enter-vr', this.saveCameraPose);
-    sceneEl.addEventListener('exit-vr', this.restoreCameraPose);
-    sceneEl.addEventListener('exit-vr', this.addHeightOffset);
-  },
+    var camera;
+    var el = this.el;
+    var sceneEl = el.sceneEl;
+    var self = this;
+
+    this.savedPose = null;
+    this.enterVRHandler = function enterVRHandler () {
+      self.saveCameraPose();
+      self.removeHeightOffset();
+    };
+    this.exitVRHandler = this.restoreCameraPose.bind(this);
```

Comment:
We need a reference to it if you want to detach the event listener later. I can move it to a method though.